### PR TITLE
iox-#2044 Refactor fixed position container and move to dust

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

I'm currently trying to replace the keep-alive mechanism from an IPC message to a timestamp in the shared memory. This is another preparatory PR which adds a `FixedPositionContainer` to replace the current stop gap solution.

I also noticed that the `dust` tests were not running in the CI and added them.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #2044

### CODE REVIEW

**Comment by elBoberido** [comment]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @elfenpiff @budrus feel free to add yourself as reviewer

During the rewrite of the `FixedPositionContainer` I noticed that there is an overlap with the `ObjectPool`. So I tried to refactor the `ObjectPool` but noticed soon that it essentially would end up in a total rewrite anyway since it uses internally pointer which prevented the use in the shared memory and other issues. Therefore I continued with the `FixedPositionContainer`. The next PR will remove the `FixedPositionContainer` from `PortPool` and also the `FixedSizeContainer` from `PortIntrospection`. It would also be possible to replace the `ObjectPool` in the gateway with the new container. I could do this also in the follow up PR.

**Review by elfenpiff — DISMISSED** [review]

File: doc/website/release-notes/iceoryx-unreleased.md [hunk 1]
```diff
@@ -162,6 +162,7 @@
 - Speed up posh tests [#1030](https://github.com/eclipse-iceoryx/iceoryx/issues/1030)
 - Roudi Environment independent from Googletest [#1533](https://github.com/eclipse-iceoryx/iceoryx/issues/1533)
 - Move test class for ctor and assignment operators to hoofs testing [#2041](https://github.com/eclipse-iceoryx/iceoryx/issues/2041)
+- Refactor `FixdePositionContainer` and move to `dust` [#2044](https://github.com/eclipse-iceoryx/iceoryx/issues/2044)
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/website/release-notes/iceoryx-unreleased.md:165]: Good catch. If there are no other findings I will fix it in one of the follow ups

**Review by elfenpiff — APPROVED** [review]

File: iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp [hunk 2]
```diff
@@ -0,0 +1,1708 @@
...
+
+    fillSutComplex();
+
+    {
+        auto it IOX_MAYBE_UNUSED = sut_complex.begin();
+        auto cit IOX_MAYBE_UNUSED = sut_complex.cbegin();
+    }
+
+    EXPECT_THAT(stats.dTor, Eq(0));
+}
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp:1704]: Wow.. These tests compete with the string tests :stuck_out_tongue_winking_eye:

Comment (elBoberido) [reply iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp:1704]: Nothing competes with the string tests :laughing:

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**